fireapp/lib/pages/settings/setting_navigation.dart

```diff
@@ -0,0 +1,11 @@
+abstract class SettingNavigation {
+  SettingNavigation._();
+
+  factory SettingNavigation.signOut() {
```

Comment:
The name of this represents the destination, not the context in which it is called, so a better name would be `login()` and `LoginSettingNavigation`. Otherwise, you have implemented the navigation code entirely correctly!